components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/nats/management/NatsEndpointManager.java

```diff
@@ -0,0 +1,105 @@
+import org.wso2.carbon.inbound.endpoint.protocol.nats.NatsInjectHandler;
+import org.wso2.carbon.inbound.endpoint.protocol.nats.NatsMessageConsumer;
+
+import java.io.IOException;
+import java.util.Properties;
+
+/**
+ * Manager which handles NATS Listeners activities for Inbound Endpoints. This is the central place to mange Http Listeners
+ * for Inbound endpoints
+ */
+public class NatsEndpointManager extends AbstractInboundEndpointManager {
+
+    private static final Log log = LogFactory.getLog(NatsEndpointManager.class);
+    private static NatsEndpointManager instance = null;
+    private NatsMessageConsumer messageConsumer = null;
+
+    public static NatsEndpointManager getInstance() {
+        if (instance == null)
+            instance = new NatsEndpointManager();
+        return instance;
+    }
+
+    /**
+     * Start NATS Listener
+     * @param port  port (not needed for NATS Inbound Endpoint)
+     * @param name  endpoint name
+     * @param inboundParameters inbound endpoint params
+     */
+    @Override public boolean startListener(int port, String name, InboundProcessorParams inboundParameters) {
+        try {
+            messageConsumer.consumeMessage();
+        } catch (IOException | InterruptedException e) {
+            log.error("An error occurred while connecting to NATS server or consuming messages. " + e);
+            messageConsumer.closeConnection();
+            return true;
+        } catch (SynapseException e) {
+            log.error("Error while retrieving or injecting NATS message. " + e.getMessage(), e);
+            return true;
+        } catch (Exception e) {
+            log.error("Error while retrieving or injecting NATS message. " + e.getMessage(), e);
+            messageConsumer.closeConnection();
+            return true;
+        }
+        return true;
+    }
+
+    /**
+     * Start NATS Inbound endpoint
+     * @param port  port (not needed for NATS Inbound Endpoint)
+     * @param name  endpoint name
+     * @param params inbound endpoint params
+     */
+    @Override public boolean startEndpoint(int port, String name, InboundProcessorParams params) {
+        Properties natsProperties = params.getProperties();
+        messageConsumer = new NatsMessageConsumer(natsProperties, name);
+        boolean sequential = true;
+        String inboundEndpointSequential = natsProperties.getProperty(PollingConstants.INBOUND_ENDPOINT_SEQUENTIAL);
+        if (inboundEndpointSequential != null) {
+            sequential = Boolean.parseBoolean(inboundEndpointSequential);
+        }
+        messageConsumer.registerHandler(
+                new NatsInjectHandler(params.getInjectingSeq(), params.getOnErrorSeq(), sequential,
+                        params.getSynapseEnvironment(), natsProperties.getProperty(NatsConstants.CONTENT_TYPE)));
+        messageConsumer.initializeMessageListener();
+        startListener(port, name, params);
+        return false;
```

Comment:
what is the boolean returned here? Shall we describe it in the javadoc.

Reply:
/**
     * Start Inbound endpoint on a particular port
     * @param port  port
     * @param name  endpoint name
     * @param inboundParameters Inbound endpoint parameters
     */
    public boolean startEndpoint(int port, String name, InboundProcessorParams inboundParameters);

It is coming from the interface and it doesn't explain the boolean returned.